phac-nml/irida-next
# Authentication: External Sign In Label Customization

## What does this PR do and why?

Adds ability to set custom name and icon for Omniauth sign in

Adds documentation on setting up authentication

Fixes #259

## Screenshots or screen recordings
Example custom text and icon on sign in page
![image](https://github.com/phac-nml/irida-next/assets/16961365/472a9dde-d7c3-4c7c-8f49-8585b139941e)

## How to set up and validate locally
1. add a `.svg` file to the `config/authentication/icons/` directory
Example File: `tyrell.svg`
![tyrell](https://github.com/phac-nml/irida-next/assets/16961365/48d6d5fc-a4f3-45a0-8b4f-1a37c7e5f003)

2. Edit `config/authenticaion/auth_config.yml` with your custom text and custom image name
```yml
development:
  omniauth_providers: [azure_activedirectory_v2]
  developer_text:
  developer_icon:
  saml_text:
  saml_icon:
  azure_activedirectory_v2_text: Tyrell Corporation
  azure_activedirectory_v2_icon: tyrell.svg
```
3. Run IRIDA Next and view the sign in page. It should look like the image above.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
